react/README.md

```diff
@@ -18,8 +18,30 @@ If you are using modules you should use `react.d.ts`, `react-dom.d.ts` or any of
 
 ## Known Problems & Workarounds
 
-### **The type of `setState` is incorrect.**
-The `setState(state)` method on `React.Component<P, S>` takes an object with a subset of the properties on `S`, but there's no way to express this in TypeScript currently. The workaround is simple: make all properties on `S` optional. There are a number of [proposals](https://github.com/Microsoft/TypeScript/issues/2710) on [ways](https://github.com/Microsoft/TypeScript/issues/4889) to [solve](https://github.com/Microsoft/TypeScript/issues/7355) this problem, but nothing seems to have been approved yet.
+### **The type of `setState` is less than ideal and a little more strict than desirable.**
+Starting with TypeScript 2.1, its more correct than it used to be, with a caveat: optional parameters on state interfaces are no longer valid.
+
+```ts
+interface FooState {
+  bar: string;
+  foo?: string;
+}
+
+const defaultFooState: FooState = {
+  bar: "Hi",
+  foo: undefined,
+};
+
+class Foo extends React.Component<{}, FooState> {
+  public doStuff() {
+    this.setState(defaultFooState);
+  }
+}
+```
+
+Will produce an error. The other way the types could be written (using `Partial<>` instead of `Pick<>`) would allow for you to set `undefined` to a parameter that is not allowed
+to be `undefined`, which could lead to bugs. Users who want to keep optional state parameters should continue to use the hack of `setState({...} as any);`.
```

Comment:
Is it true that this will be fixed by https://github.com/Microsoft/TypeScript/pull/12826? If so, can we note that here, and/or in a comment inline?